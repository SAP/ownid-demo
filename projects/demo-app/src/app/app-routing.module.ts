import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { NotesGuard } from './services/notes-guard.service';
import { LoginGuard } from './services/login-guard.service';


const routes: Routes = [
  {
    path: 'login',
    loadChildren: () => import('./pages/login/login.module').then((m) => m.LoginModule),
    canActivate: [LoginGuard],
  },
  {
    path: 'instant-login',
    loadChildren: () => import('./pages/instant-login/instant-login.module').then((m) => m.InstantLoginModule),
    canActivate: [LoginGuard],
  },
  {
    path: 'forgot-password',
    loadChildren: () => import('./pages/forgot-password/forgot-password.module').then((m) => m.ForgotPasswordModule),
    canActivate: [LoginGuard],
  },
  {
    path: 'reset-password',
    loadChildren: () => import('./pages/reset-password/reset-password.module').then((m) => m.ResetPasswordModule),
    canActivate: [LoginGuard],
  },
  {
    path: 'register',
    loadChildren: () => import('./pages/registration/registration.module').then((m) => m.RegistrationModule),
    canActivate: [LoginGuard],
  },
  {
    path: 'instant-register',
    loadChildren: () => import('./pages/instant-registration/instant-registration.module').then((m) => m.InstantRegistrationModule),
    canActivate: [LoginGuard],
  },
  {
    path: 'notes',
    loadChildren: () => import('./pages/notes/notes.module').then((m) => m.NotesModule),
    canActivate: [NotesGuard],
  },
  {
    path: 'passwordless',
    loadChildren: () => import('./pages/passwordless/passwordless.module').then((m) => m.PasswordlessModule),
  },
  {
    path: '',
    loadChildren: () => import('./pages/home/home.module').then((m) => m.HomeModule),
    canActivate: [LoginGuard],
  },
  {
    path: '**',
    redirectTo: '',
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule {}
