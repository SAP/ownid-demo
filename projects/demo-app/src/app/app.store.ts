import { BehaviorSubject } from 'rxjs';

export interface INote {
  characters: number;
  words: number
  id: string;
  body: string;
  created: string;
  updated: string;
}

export interface IProfile {
  email: string;
  name: string;
}

export class AppStore {
  readonly profile$ = new BehaviorSubject<IProfile>({} as IProfile);

  readonly notes$ = new BehaviorSubject<INote[]>([]);

  readonly isOwnidUser$ = new BehaviorSubject<boolean>(false);
}
