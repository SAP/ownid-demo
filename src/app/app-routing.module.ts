import { NgModule } from "@angular/core";
import { Routes, RouterModule } from "@angular/router";
import { UserGuardService } from "@services/user-guard.service";

const routes: Routes = [
  {
    path: "login",
    loadChildren: () =>
      import("./pages/sign/sign-page.module").then((m) => m.SignPageModule)
  },
  {
    path: 'forgot-password',
    loadChildren: () => import('./pages/forgot-password/forgot-password.module').then((m) => m.ForgotPasswordModule),
  },
  {
    path: 'reset-password',
    loadChildren: () => import('./pages/reset-password/reset-password.module').then((m) => m.ResetPasswordModule),
  },
  {
    path: "account",
    loadChildren: () =>
      import("./pages/user/user-page.module").then((m) => m.UserPageModule),
    canActivate: [UserGuardService]
  },
  {
    path: "",
    pathMatch: "full",
    redirectTo: "account"
  },
  {
    path: "**",
    redirectTo: "account"
  }
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule {}
