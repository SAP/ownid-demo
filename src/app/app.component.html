<mat-toolbar class="toolbar">
  <header class="header">
    <span class="headline">Sample app</span>
    <span>
      <button mat-button>About</button>
      <button mat-button>Services</button>
      <button mat-button>Portfolio</button>
      <button mat-button>Contact</button>
    </span>
  </header>
</mat-toolbar>

<router-outlet></router-outlet>

<mat-toolbar class="footer">
  <span>Copyright Â© 2020 - Sample app</span>
</mat-toolbar>
