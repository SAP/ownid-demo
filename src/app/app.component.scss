.toolbar {
  display: block;
  @media (max-width: 599px) {
    height: 96px;
  }
}

.header {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  height: 64px;
}

.footer {
  display: flex;
  justify-content: center;
  font-size: 12px;
  
  .row {
    height: 20px;
  }
}

@media (max-width: 599px) {
  .headline {
    margin-top: 20px;
  }
}

@media (min-width: 599px) {
  .header {
    flex-direction: row;

    max-width: 1100px;
    height: 64px;
    margin: 0 auto;
  }
}

