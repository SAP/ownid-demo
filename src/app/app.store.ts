import { BehaviorSubject } from 'rxjs';
import { Injectable } from '@angular/core';

export interface IProfile {
  email: string;
  nickname: string;
  firstName: string;
  lastName: string;
}

@Injectable()
export class AppStore {
  readonly profile$ = new BehaviorSubject<IProfile>({} as IProfile);
}
