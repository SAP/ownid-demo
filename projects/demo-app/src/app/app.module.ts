import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { GigyaService } from './services/gigya.service';
import { NotesGuard } from './services/notes-guard.service';
import { LoginGuard } from './services/login-guard.service';
import { AppStore } from './app.store';

@NgModule({
  declarations: [
    AppComponent,
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    HttpClientModule,
  ],
  providers: [GigyaService, NotesGuard, LoginGuard, AppStore],
  bootstrap: [AppComponent]
})
export class AppModule {}
