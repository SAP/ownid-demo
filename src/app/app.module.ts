import { BrowserModule } from "@angular/platform-browser";
import { NgModule, ErrorHandler } from "@angular/core";

import { BrowserAnimationsModule } from "@angular/platform-browser/animations";
import { MatToolbarModule } from "@angular/material/toolbar";
import { MatButtonModule } from "@angular/material/button";
import { MatIconModule } from "@angular/material/icon";
import { UserGuardService } from "@services/user-guard.service";
import { GigyaService } from "@services/gigya.service";
import { AppRoutingModule } from "./app-routing.module";
import { AppComponent } from "./app.component";
import { GlobalErrorHandler } from "./infrastructure/global-error-handler";
import { AppStore } from "./app.store";

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    MatToolbarModule,
    MatButtonModule,
    MatIconModule
  ],
  providers: [
    GigyaService,
    {
      provide: ErrorHandler,
      useClass: GlobalErrorHandler
    },
    AppStore,
    UserGuardService
  ],
  bootstrap: [AppComponent]
})
export class AppModule {}
