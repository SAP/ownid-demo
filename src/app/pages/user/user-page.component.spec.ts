import { async, TestBed } from "@angular/core/testing";
import { NgZone, NO_ERRORS_SCHEMA } from "@angular/core";
import { GigyaService } from "@services/gigya.service";
import { AppStore } from "../../app.store";
import { UserPageComponent } from "./user-page.component";
import { SetProfileCommand } from "../sign/commands/set-profile.command";

describe("UserPageComponent", () => {
  let setProfileCommand: SetProfileCommand;
  let gigyaService: GigyaService;

  beforeEach(() => {
    setProfileCommand = {} as SetProfileCommand;
    setProfileCommand.execute = jest.fn();
    gigyaService = {} as GigyaService;
    gigyaService.getProfile = jest
      .fn()
      .mockImplementation((c) => c({ profile: {} }));
  });

  describe("Snapshot", () => {
    beforeEach(async(() => {
      TestBed.configureTestingModule({
        providers: [
          AppStore,
          {
            provide: SetProfileCommand,
            useValue: setProfileCommand
          },
          {
            provide: GigyaService,
            useValue: gigyaService
          }
        ],
        declarations: [UserPageComponent],
        schemas: [NO_ERRORS_SCHEMA]
      }).compileComponents();
    }));

    test("check snapshot", () => {
      const fixture = TestBed.createComponent(UserPageComponent);

      expect(fixture).toMatchSnapshot();
    });
  });

  describe("onLink", () => {
    it("should update linked$ with true", () => {
      const appStore = new AppStore();
      const sut = new UserPageComponent(
        appStore,
        new SetProfileCommand(appStore),
        gigyaService,
        new NgZone({})
      );
      sut.onLink();
      expect(sut.linked$.value).toBeTruthy();
    });
  });

  describe("logout", () => {
    it("should call gigyaService.logout", () => {
      const appStore = new AppStore();
      const sut = new UserPageComponent(
        appStore,
        new SetProfileCommand(appStore),
        gigyaService,
        new NgZone({})
      );
      gigyaService.logout = jest.fn();
      sut.onLogout();
      expect(gigyaService.logout).toBeCalled();
    });
  });
});
