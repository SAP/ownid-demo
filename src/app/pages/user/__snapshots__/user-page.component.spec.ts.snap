// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPageComponent Snapshot check snapshot 1`] = `
<app-user-page
  gigyaService={[Function Object]}
  linked$={[Function BehaviorSubject]}
  ngZone={[Function NgZone]}
  profile$={[Function BehaviorSubject]}
  setProfileCommand={[Function Object]}
  store={[Function AppStore]}
>
  <section
    class="sign-body"
  >
    <div
      class="container"
    >
      
      <div />
    </div>
  </section>
</app-user-page>
`;
